pkg/gornir/runner.go

```diff
@@ -1,43 +1,68 @@
 package gornir
 
 import (
+	"context"
 	"sync"
 )
 
+type TaskParameters struct {
```

Comment:
I'd try to separate the `hosts` and `task` parameters or not have the Host as part of this struct to decouple the two (will make testing simpler). Also is confusing to have a Host here, while Gornir has a list of Hosts as well.

Reply:
I will add a separate issue to address the host portion, I think it will be better we reference an interface (behavior) instead  of a concrete type (implementation details).

Reply:
> I'd try to separate the hosts and task parameters or not have the Host

The purpose of `TaskParameters` is twofold:

1. Simplify the signature and make things easier to pass around
2. Allow us to add new fields to it that may not be mandatory without breaking code

So having the host is convenient (even though I agree it's ugly-ish).

> is confusing to have a Host here, while Gornir has a list of Hosts as well

Yeah, thinking about it the Gornir object should go away from that structure.

> decouple the two (will make testing simpler)

Shouldn't be much different, in the end it's just a matter of calling a function either like `Run(ctx, host...)` or `Run(ctx, TaskParameters{host: host, ...}, ...)`

Overall I think this part is the trickiest from an API design perspective as I don't want to have a superlong/complex signature that may have to be changed otherwise but I agree combining "bare" arguments with `TaskParameters` is not the most elegant solution.

I think I need to think more about this.

As I think there is value in the decoupling of the context, worst case we could go with this solution (after addressing your other feedback) and open a new issue to discuss more about this.